src/libsyntax/ast.rs

```diff
@@ -46,7 +45,7 @@ pub struct SyntaxContext(pub u32);
 /// An identifier contains a Name (index into the interner
 /// table) and a SyntaxContext to track renaming and
 /// macro expansion per Flatt et al., "Macros That Work Together"
-#[derive(Clone, Copy, Eq)]
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
It shouldn't make a difference since we [`panic!`ed](https://github.com/rust-lang/rust/pull/34570/files#diff-0de6e63ecfba5c83fb06c9e981f292a6L116) before when using `==` to compare two identifiers with different syntax contexts.